tracer/src/Datadog.Trace.MSBuild/DatadogLogger.cs

```diff
@@ -174,29 +175,36 @@ private void EventSource_ProjectStarted(object sender, ProjectStartedEventArgs e
 
                 string projectName = Path.GetFileName(e.ProjectFile);
 
-                Span projectSpan = _tracer.StartSpan(BuildTags.BuildOperationName, parent: parentSpan.Context);
+                string targetName = string.IsNullOrEmpty(e.TargetNames) ? "build" : e.TargetNames?.ToLowerInvariant();
+                Span projectSpan = _tracer.StartSpan($"msbuild.{targetName}", parent: parentSpan.Context);
 
                 if (projectName != null)
                 {
                     projectSpan.ServiceName = projectName;
                 }
 
-                projectSpan.ResourceName = projectName;
                 projectSpan.SetTraceSamplingPriority(SamplingPriorityValues.AutoKeep);
                 projectSpan.Type = SpanTypes.Build;
 
+                string targetFramework = null;
                 foreach (KeyValuePair<string, string> prop in e.GlobalProperties)
                 {
                     projectSpan.SetTag($"{BuildTags.ProjectProperties}.{prop.Key}", prop.Value);
+                    if (string.Equals(prop.Key, "TargetFramework", StringComparison.OrdinalIgnoreCase))
+                    {
+                        targetFramework = prop.Value;
+                    }
                 }
 
+                projectSpan.ResourceName = (string.IsNullOrEmpty(e.TargetNames) ? "Build" : e.TargetNames) + $"/{targetFramework}";
```

Comment:
Same here, could we have multiple target names? I can't remember how it works with `TargetFrameworks` (multiple), but I presume it's essentially one span per `TargetFramework`?

Reply:
Yes, that's the behaviour I'm seeing... one span per `TargetFramework`